### PR TITLE
planner/core: fix nil pointer dereference tidb_decode_key()

### LINKED ISSUE #33015: nil pointer dereference in tidb_decode_key()
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
select tidb_decode_key( '74800000000000012B5F72800000000000A5D3' );
```

### 2. What did you expect to see? (Required)

No panic.

### 3. What did you see instead (Required)

```
ERROR 1105 (HY000): runtime error: invalid memory address or nil pointer dereference
```

### 4. What is your TiDB version? (Required)

328a8bcab

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #33015

### What is changed and how it works?

The result of `TableByID` is not checked:

https://github.com/pingcap/tidb/blob/5170f2f414808a8b6e91a290807287b536ddddf1/planner/core/expression_rewriter.go#L2066-L2075

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: expression/integration_test.go [hunk 1]
```diff
@@ -2816,7 +2816,7 @@ func TestTiDBDecodeKeyFunc(t *testing.T) {
 	result.Check(testkit.Rows("7480000000000000FF2E5F728000000011FFE1A3000000000000"))
 	warns := tk.Session().GetSessionVars().StmtCtx.GetWarnings()
 	require.Len(t, warns, 1)
-	require.Error(t, warns[0].Err, "invalid record/index key: 7480000000000000FF2E5F728000000011FFE1A3000000000000")
+	require.Error(t, warns[0].Err, "invalid key: 7480000000000000FF2E5F728000000011FFE1A3000000000000")
```
Comment (hawkingrei) [inline expression/integration_test.go:2819]: use ```require.EqualError``` ？

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by tisonkun — CHANGES_REQUESTED** [review]
Sorry. Test failure occurs on related changes:

```
[2022-03-12T09:02:29.852Z]     result.go:50: 
[2022-03-12T09:02:29.852Z]         	Error Trace:	result.go:50
[2022-03-12T09:02:29.852Z]         	            				integration_test.go:2802
[2022-03-12T09:02:29.852Z]         	Error:      	Not equal: 
[2022-03-12T09:02:29.852Z]         	            	expected: "[{\"handle\":\"{1.1, 11}\",\"table_id\":50}]\n"
[2022-03-12T09:02:29.852Z]         	            	actual  : "[7480000000000000325f7205bff199999999999a013131000000000000f9]\n"
[2022-03-12T09:02:29.852Z]         	            	
[2022-03-12T09:02:29.852Z]         	            	Diff:
[2022-03-12T09:02:29.852Z]         	            	--- Expected
[2022-03-12T09:02:29.852Z]         	            	+++ Actual
[2022-03-12T09:02:29.852Z]         	            	@@ -1,2 +1,2 @@
[2022-03-12T09:02:29.852Z]         	            	-[{"handle":"{1.1, 11}","table_id":50}]
[2022-03-12T09:02:29.852Z]         	            	+[7480000000000000325f7205bff199999999999a013131000000000000f9]
[2022-03-12T09:02:29.852Z]         	            	 
[2022-03-12T09:02:29.852Z]         	Test:       	TestTiDBDecodeKeyFunc
[2022-03-12T09:02:29.852Z]         	Messages:   	sql:select tidb_decode_key( '7480000000000000325f7205bff199999999999a013131000000000000f9' ), args:[]
```

**Review by hawkingrei — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Pull request merged**